src/components/dashboard/browse/BeanItemDetailsContainer.js

```diff
@@ -27,17 +28,45 @@ class BeanItemDetailsContainer extends Component {
     update(page, reset, id) {
         const {dispatch} = this.props;
 
+        if (this.props.bean[id]) {
+            this.loadRoaster(page, reset, this.props.bean[id].roasterId);
+            return;
+        }
+
         dispatch(getItem(id)).then(() => {
-           const limit = 10;
-            let offset = (page - 1) * limit;
-            dispatch(getRoasterItems(this.props.bean.roasterId, reset ? 0 : offset, limit));
+            this.loadRoaster(page, reset, this.props.bean[id].roasterId);
         });
     }
 
+    bean() {
+        return this.props.bean[this.props.params.id];
+    }
+
+    loadRoaster(page, reset, id) {
+        const {dispatch} = this.props;
+        const limit = 10;
+        const offset = (page - 1) * limit;
+
+        if (!this.props.roasters[id]) {
+            dispatch(getRoaster(id));
+        }
+        dispatch(getRoasterItems(id, reset ? 0 : offset, limit));
+    }
+
+    roaster() {
+        if (!this.bean()) {
+            return {};
+        }
+
+        return this.props.roasters[this.bean().roasterId];
```

Comment:
same goes for roaster. it's actually a set of roaster object that we can retrieve by id.